### DISCUSSION TITLE
desktop switcher - - wanting to kick over the limitations

### DISCUSSION THREAD

**Discussion by dabeegmon** [lxqt/lxqt#2401] [Q&A]
I'm going to start with a brief picture of my system (I think it will help with the request). 
I'm running on a desktop with LOTS of graphics resources - - - 4 - 24" monitors + one 4k (43") monitor
Have found myself using desktops to separate out working areas - - - means I use different desktops to do different things. Lots of different things! 
Am running into some issues though. 
1. limitation - - - - max 36 desktops - - - - why is there a limitation? - - - I presently have a 40 desktop (desktop switcher) open and with 18 characters in the 'name' I would have room for more - - - - like maybe 46 (under the 'preferences' in the program launcher is where only 36 lines are allowed (in 'Workspaces'). 
2.  the 'names' on the desktops are not saved so that if a reboot is required - - - - well I need to re-enter all the names. (that's tedious and should really be necessary. 

I'm using xfwm4 for a window manager - - - that may have some bearing on this area. 
Have spent some couple hours looking through source code trying to see if I could find where this hard coding is. Sadly (I'm not at all a programmer) I can't find enough bread crumbs to get me into the right area. 

I understand that most of the world thinks that a laptop is the bee's knees but I'm finding that even with all my screen resources I'm occasionally short of space to put everything I'm using on 'one' desktop - - -  (I've been considering adding 2 more 4k monitors and ditching the 24" ones that would increase my screen real estate by 50%) but the hard coded choices are becoming ever more of a block on my ability to use all of my resources. 

Possibly its only 1 location where I could modify the code for only my computer - - - then maybe someone could point out where that code is hiding. My guess is that this is twined with at least 2 if not 4 other things and its just not so easy. 

If this hasn't been clear - - - - please advise soonest with questions and I'll try to improve my description.

TIA

**Comment by tsujan** [comment]
> limitation - - - - max 36 desktops

There's no limitation in the [code](https://github.com/lxqt/lxqt-panel/tree/master/plugin-desktopswitch).

> the 'names' on the desktops are not saved

A known problem in `xfwm4`.

**Comment by dabeegmon** [comment]
Thank you for your response!! 

So if there isn't a limitation in the 'code' is the limitation in xfwm4 (along with the names issue)?

**Comment by tsujan** [comment]
Logically, there can be three sources of problems: (1) our code; (2) WM codes; or (3) `KWindowSystem`, which we use for such jobs.

LXQt's code is simple in this regard — it mostly uses `KWindowSystem` — so, most probably, the problem is either in WM or `KWindowSystem`.

You could check the first by trying another WM, like Openbox (only If you have a single monitor — it has bugs with multi-screen setups) or KWin (good with multi-screen setups).

**Comment by dabeegmon** [comment]
Thanking you for your assistance. 

Will dig further - - - - if I can find the smoking gun - - - - should I report back?

**Comment by tsujan** [comment]
Also, a clear screenshot of the desktop switcher with 40 desktops might help (perhaps the first one in the history of mankind). I wonder how one could fit 46 characters in each cell! It'll need a screen whose width is much more than 7400 pixels (with an imaginary 4-pixel wide font).

> should I report back?

Why not?

**Comment by dabeegmon** [comment]
(Grin) dunno where you're getting 46 characters in each cell - - - - I'm using 18 characters (ie '20 parxxxplanning' is one label') so there is a little bit of white space before and after the text and then the 'button' outline - - - I have things arranged in 2 rows of 20 and could - - - better I HAD 40 desktops in DesktopSwitcher - - - now back to 36 .  I wouold have room for perhaps 2 x 23  - - - - I do have 3840 pixels to play with.  (looks like a 10 point font to me) 

I spent a little time looking into the KWindowSystem - - - - do you know where the specific stuff used for the workspaces hides out at? (Can you tell I'm a noob at this?)

**Comment by tsujan** [comment]
> dunno where you're getting 46 characters

By misreading your comments ("with 18 characters in the 'name' I would have room for more - - - - like maybe 46") and expecting big numbers.

> do you know where the specific stuff

No. But I remember that the maximum number of virtual desktops was set to 20 in KDE a long time ago. To me, that's reasonable (if not beyond reasonable) when I consider the maximum number of F keys . Anyway, since `KWindowSystem` is made by KDE devs, I wonder why they might have supported 36 there.

BTW, I'm converting this to a discussion...
